crates/subspace-networking/src/constructor.rs

```diff
@@ -239,7 +240,9 @@ pub struct Config<LocalRecordProvider> {
     /// Backoff policy for temporary banning of unreachable peers.
     pub temporary_ban_backoff: ExponentialBackoff,
     /// Optional external prometheus metrics. None will disable metrics gathering.
-    pub metrics: Option<Metrics>,
+    pub external_metrics: Option<Metrics>,
+    /// Internal prometheus metrics. None will disable metrics gathering.
+    pub metrics: Option<SubspaceMetrics>,
```

Comment:
What do you think about renaming `external_metrics: Option<Metrics>` to `libp2p_metrics: Libp2pMetrics`? Then the rest can be called just `metrics: Option<Metrics>` since it is clearly ours?